dashboard/lib/tasks/setup_or_migrate.rake

```diff
@@ -9,8 +10,10 @@ namespace :db do
     true
   end
 
-  task :setup_or_migrate => [:environment, :load_config] do
-    if database_exists?
+  task :setup_or_migrate do
+    db_exists = database_exists?
+    Rake::Task["db:load_config"].invoke
```

Comment:
I assume we don't have direct control over the `environment` task here - if we did, I'd say the _most_ correct thing to do would be make _it_ depend on some kind of `ensure_db_exists` task, since you've already identified that it depends on the db existing.